src/action/filesystems.rs

```diff
@@ -102,3 +165,92 @@ impl super::Response for Response {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::path::PathBuf;
+    use users::{get_current_uid, get_current_gid};
+
+    #[test]
+    fn test_is_any_filesystem_exist() {
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+        assert_ne!(session.reply_count(), 0);
+    }
+
+    /// Returns all responses whose `source` is equal to `fs_name`.
+    fn find_filesystems_by_name<'a>(
+        session: &'a session::test::Fake,
+        fs_name: &'a PathBuf,
+    ) -> Vec<&'a Response> {
+        let mut responses = Vec::new();
+
+        for i in 0..session.reply_count() {
+            let response: &Response = session.reply(i);
+            if response.mount_info.source == *fs_name {
+                responses.push(response);
+            }
+        }
+
+        responses
+    }
+
+    /// Unit-like struct, representing a filesystem for testing with `fuse`.
+    struct FuseFilesystem;
+
+    impl fuse::Filesystem for FuseFilesystem {}
+
+    #[test]
+    fn test_fuse_filesystem() {
+        let fs_name = PathBuf::from("fuse-test-fs");
+
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let mountpoint = tmp_dir.path();
+
+        let fs_name_option = format!("fsname={}", fs_name.to_str().unwrap());
+        let options = [
+            "-o", "ro",
+            "-o", "nosuid",
+            "-o", "nodev",
+            "-o", "relatime",
+            "-o", &fs_name_option,
+        ];
+        let options = options.iter().map(|opt| opt.as_ref())
+            .collect::<Vec<&std::ffi::OsStr>>();
+
+        // Spawn a background thread to handle filesystem operations.
+        // When `_background_session` is dropped, filesystem will be unmounted.
+        let _background_session = unsafe {
+            fuse::spawn_mount(FuseFilesystem, &mountpoint, &options).unwrap()
+        };
+
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+
+        let fuse_mounted_fs = find_filesystems_by_name(&session, &fs_name);
+        assert_eq!(fuse_mounted_fs.len(), 1);
+
+        let mount_info = &fuse_mounted_fs[0].mount_info;
+        assert_eq!(mount_info.source, fs_name);
+        assert_eq!(mount_info.dest, tmp_dir.path());
+        assert_eq!(mount_info.fstype, "fuse");
```

Comment:
Is this value guaranteed to be set to `fuse` or is it a FUSE library implementation detail? In the latter case there might be a way to castomize it which would be a preferred decision since it is not a good idea to rely on some accidental constants in tests.

Reply:
This is guaranteed by `FUSE`. Here is a quote from man:
```
subtype=TYPE
              Sets the filesystem type (third field in /etc/mtab). The
              default is the name of the filesystem process. If the kernel
              suppports it, /etc/mtab and /proc/mounts will show the
              filesystem type as fuse.TYPE

              If the kernel doesn't support subtypes, the source filed will
              be TYPE#NAME, or if fsname option is not specified, just TYPE.
```

Anyway, I think I'll customize it just in case. But there is one problem - the last paragraph in the quote. Should we care about that?

Reply:
Since this is just a test, I think we can assume that whoever runs it is expected to have relatively new kernel that supports these subtypes. If this assumption proves to be incorrect, we can always rethink this approach, but for the time being, it looks good to me.